File: Number of Subarrays of 0's[17-Mar-2023]/Solution.java
//			>> JAVA CODE <<

class Solution{
	long no_of_subarrays(int N, int [] arr) {
		
		long ans=0;
		long cnt=0;

		
		//just count the continuous number of 0's
		for(var in:arr)
		{
		    if(in==0)
		    {
		        cnt++;
		    }
		    else
		    {
				// number of subarrays = n*(n+1)/2;
		        ans+=((long)(cnt*(cnt+1))/2);
		        cnt=0;
		    }
		}
		if(cnt!=0)
		{
		    ans+=((long)(cnt*(cnt+1))/2);
		}
		
		    return ans;
	}
}

File: Number of Subarrays of 0's[17-Mar-2023]/solution.cpp
//			>> C++ CODE <<

long long int no_of_subarrays(int n, vector<int> &arr) {
    	  long ans = 0;
        long cnt = 0;

	  // count the continuous 0's
        for(int i = 0; i < n; i++) {
            if(arr[i] == 0) {
                cnt++;
            } else {
                ans += ((long)(cnt*(cnt+1))/2);
                cnt = 0;
            }
        }

        if(cnt != 0) {
            ans += ((long)(cnt*(cnt+1))/2);
        }

        return ans;
}